src/Sentry/Internal/SettingLocator.cs

```diff
@@ -35,19 +35,16 @@ public string GetDsn()
         // For DSN only
         // An empty string set on the option should NOT be converted to null because it is used
         // to indicate the the SDK is disabled.
-
-        var dsn = _options.Dsn;
-        if (dsn != null)
+        _options.Dsn ??= GetEnvironmentVariable(Constants.DsnEnvironmentVariable)
+                         ?? AssemblyForAttributes?.GetCustomAttribute<DsnAttribute>()?.Dsn;
+        if (_options.Dsn is null)
         {
-            return dsn;
+            throw new ArgumentNullException(
```

Comment:
Also, kind of unrelated, if we can use the new `ArgumentNullException.ThrowIfNull();` and friends. We'll help JIT emit faster code.

Reply:
I don't think we can use `ArgumentNullException.ThrowIfNull()` as it doesn't allow you to pass an exception message (only the name of the parameter that was null).